app/src/main/java/acr/browser/lightning/utils/Utils.java

```
@@ -309,4 +312,20 @@ public static void drawTrapezoid(Canvas canvas, int color, boolean withShader) {
         canvas.drawPath(wallpath, paint);
     }
 
+    /**
+     * Calculate the SHA-1 hash of the given string
+     * @param str   the string to hash
+     * @return      the long representation of the hash
+     */
+    public static long hash(final String str) {
```

...but practically swallowing an exception - not so much. This probably should "never happens", but I would rethrow RuntimeException instead of nearly silently generating many hash collisions.